sunpy/instr/goes.py

```diff
@@ -583,16 +698,533 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+                      date=str(goeslc.data.index[0]))
+    # Enter results into new version of GOES LightCurve Object
+    # Use copy.deepcopy for replicating meta and data so that input
+    # lightcurve is not altered.
+    lc_new = lightcurve.LightCurve(meta=copy.deepcopy(goeslc.meta),
+                                   data=copy.deepcopy(goeslc.data))
+    lc_new.data["luminosity_xrsa"] = lx_out["shortlum"].to("W").value
+    lc_new.data["luminosity_xrsb"] = lx_out["longlum"].to("W").value
+
+    return lc_new
+
+def _goes_lx(longflux, shortflux, obstime=None, date=None):
+    """
+    Calculates solar X-ray luminosity in GOES wavelength ranges.
+
+    This function calculates the X-ray luminosity from the Sun in the
+    GOES wavelength ranges (1-8 angstroms and 0.5-4 angstroms) based
+    on the observed GOES fluxes.  The units of the results are erg/s.
+    The calculation is made by simply assuming that the radiation is
+    emitted isotropically, i.e. is distributed over a spherical
+    surface area with a radius equal to the Sun-Earth distance.
+
+    Parameters
+    ----------
+    longflux : `~astropy.units.Quantity`
+        Array containing the observed GOES/XRS long channel flux.
+        Units=[W/m**2]
+
+    shortflux : `~astropy.units.Quantity`
+        Array containing the observed GOES/XRS short channel flux.
+        Units=[W/m**2]
+
+    obstime : (optional) array-like of datetime objects
+        Measurement times corresponding to each flux measurement.
+        Assumes each pair of 0.5-4 and 1-8 angstrom flux measurements
+        were taken simultaneously.
+
+    date : (optional) datetime object or valid date string.
+        Date at which measurements were taken.
+
+    Returns
+    -------
+    lx_out : dictionary
+        dictionary containing the following fields.
+        longlum : astropy.units.Quantity
+            Array of luminosity in the 1-8 angstroms range.
+
+        shortlum : astropy.units.Quantity
+            Array of luminosity in the 0.5-4 angstroms range.
+
+        longlum_int : (only present if obstime kwarg is set)
+
+        shortlum_int : (only present if obstime kwarg is set)
+
+    Notes
+    -----
+    This function calls _calc_xraylum() to calculate luminosities.
+    For more information on how this is done, see docstring of that
+    function.
+
+    Examples
+    --------
+    >>> from datetime import datetime
+    >>> from astropy.units.quantity import Quantity
+    >>> longflux = Quantity([7e-6,7e-6,7e-6,7e-6,7e-6,7e-6], unit='W/m**2')
+    >>> shortflux = Quantity([7e-7,7e-7,7e-7,7e-7,7e-7,7e-7], unit='W/m**2')
+    >>> obstime = np.array([datetime(2014,1,1,0,0,0),
+                            datetime(2014,1,1,0,0,2),
+                            datetime(2014,1,1,0,0,4),
+                            datetime(2014,1,1,0,0,6),
+                            datetime(2014,1,1,0,0,8),
+                            datetime(2014,1,1,0,0,10),], dtype=object)
```

Comment:
this can be done using numpy as commented in the tests
```python
obstime = np.arange('2014-01-01 00:00:00','2014-01-01 00:00:11', 2, dtype='datetime64[s]')
```